### ISSUE TITLE
Unable to make extention work in Cursor IDE

### ISSUE THREAD

**Issue by mrjxtr** [be5invis/vscode-custom-css#245]
![image](https://github.com/user-attachments/assets/23d9e140-f8af-4265-854d-936deea0826a)

Additional Info:
1. I have a dual boot system with Windows and vanilla Arch.
1. I was able to make this work seamlessly on Windows.
1. I initially encountered the same issue when using this with VScode but was fixed easily with:
    - `sudo chown -R $(whoami) "$(which code)"`
    - `sudo chown -R $(whoami) /opt/visual-studio-code`
    - `sudo chown -R $(whoami) /usr/bin/code`
1. I used `yay -S cursor-bin` to install cursor.

What I tried:
1. I tried to do the same with cursor, using `which cursor` and `whereis cursor` to locate cursor directory.
1. I also did `sudo chown -R $(whoami) /opt/cursor-bin`

I understand that Cursor is an AppImage for linux and maybe that is the issue? but is there a fix?

**Comment by joshmfrankel** [comment]
I've run into this same issue as well. What I've tried (that has yet to be successful):

1. Run app image with sudo `sudo ./cursor-0.xxxxx.AppImage --no-sandbox`
2. Run with custom user directory `./cursor-0.xxxxx.AppImage --user-data-dir=~/Applications/Cursor` since the directory will have full permission control
3. Extract AppImage files to a writable location (I couldn't get this to run successfully)
```
./cursor-0.xxxxx.AppImage --appimage-extract
./squashfs-root/AppRun
```

All of the above encounter the same issue as the screenshot above. I believe the problem is that the mounted file system is marked as `ro` read-only. 

I tried `mount | grep cursor` and found the directory mentioned in the error message as read-only. 

```
❯ mount | grep cursor
~/cursor-0.xxxxx.AppImage on /tmp/.mount_cursorl9krmK type fuse.cursor-0.43.6x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=0,group_id=0)
```

Tried to remount it with `sudo mount -o remount,rw ~/cursor-0.xxxxx.AppImage` and it changes to read-write. However, after restarting Cursor to enact the change, it reverts to read-only. 🤷🏻

**Comment by resslr** [comment]
> ![image](https://private-user-images.githubusercontent.com/176251348/387928364-23d9e140-f8af-4265-854d-936deea0826a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY2NDQ2NDQsIm5iZiI6MTczNjY0NDM0NCwicGF0aCI6Ii8xNzYyNTEzNDgvMzg3OTI4MzY0LTIzZDllMTQwLWY4YWYtNDI2NS04NTRkLTkzNmRlZWEwODI2YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExMlQwMTEyMjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YzI0Y2FjNmEyYzZlZGU1MGMyYTk0NTRlZjYxZmE3ZjVmYmNiN2I4Yjk3MzczNGM0OTFlZjliOTdjYWY1NjgzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._M62LjKbvjZR-6YRDYjVkEjvXkhYgKmj09viF6T_InU)
> 
> Additional Info:
> 
> 1. I have a dual boot system with Windows and vanilla Arch.
> 2. I was able to make this work seamlessly on Windows.
> 3. I initially encountered the same issue when using this with VScode but was fixed easily with:
>    
>    * `sudo chown -R $(whoami) "$(which code)"`
>    * `sudo chown -R $(whoami) /opt/visual-studio-code`
>    * `sudo chown -R $(whoami) /usr/bin/code`
> 4. I used `yay -S cursor-bin` to install cursor.
> 
> What I tried:
> 
> 1. I tried to do the same with cursor, using `which cursor` and `whereis cursor` to locate cursor directory.
> 2. I also did `sudo chown -R $(whoami) /opt/cursor-bin`
> 
> I understand that Cursor is an AppImage for linux and maybe that is the issue? but is there a fix?

Can you walk me through your vscode rice 😭 it looks so good + a link to your dotfiles

**Comment by mrjxtr** [comment]
I have them all on my public repos... Although I think they need to be updated because I made some changes since this post. You can have a look here

**Comment by resslr** [comment]
Thank you!

**Comment by resslr** [comment]
> I have them all on my public repos... Although I think they need to be updated because I made some changes since this post. You can have a look at my [dotfiles](https://github.com/mrjxtr/dotfiles-arch) and [custom configs](https://github.com/mrjxtr/configs-arch)

Btw was this issue fixed for you?

**Comment by mrjxtr** [comment]
@resslr No, they haven't even responded on this issue in Github yet.

**Comment by be5invis** [comment]
Looks like related to sandboxing? This extension **REQUIRES** modifying VSCode's files, otherwise the custom CSS simply won't work since VSCode has no way to load it.

**Comment by mrjxtr** [comment]
> Looks like related to sandboxing? This extension **REQUIRES** modifying VSCode's files, otherwise the custom CSS simply won't work since VSCode has no way to load it.

Yeah, I needed to change permissions on some directories to make this work with VS Code...but the problem is it's different for **CURSOR** probably because cursor is using an Appimage and and is using sandboxing.. I was able to make it work for VS Code but not for Cursor

**Comment by keyding** [comment]
I have submitted PR: https://github.com/be5invis/vscode-custom-css/pull/248

It should be the wrong file to modify, I tried again in my cursor and it works.

**Comment by mrjxtr** [comment]
> I have submitted PR: #248
> 
> It should be the wrong file to modify, I tried again in my cursor and it works.

@keyding  Hopefully they will merge this... But if I want to test this, I just need to create a fork, add the lines you added, build the app and it should work?

**Comment by keyding** [comment]
> > I have submitted PR: #248
> > It should be the wrong file to modify, I tried again in my cursor and it works.
> 
> @keyding Hopefully they will merge this... But if I want to test this, I just need to create a fork, add the lines you added, build the app and it should work?

Yes, I've tested it and it's fine. I'm using macOS.

**Comment by keyding** [comment]
@mrjxtr 

Or you can just modify the source code of the extension.

First, go to the extension directory

```sh
cd ~/.cursor/extensions/be5invis.vscode-custom-css-7.4.0/src
```

Edit `extension.js`, add the following code

```js
const vscode = require("vscode");
const fs = require("fs");
const os = require("os");
const path = require("path");
const msg = require("./messages").messages;
const uuid = require("uuid");
const fetch = require("node-fetch");
const Url = require("url");

function activate(context) {
	const appDir = require.main
		? path.dirname(require.main.filename)
		: globalThis._VSCODE_FILE_ROOT;
	if (!appDir) {
		vscode.window.showInformationMessage(msg.unableToLocateVsCodeInstallationPath);
	}

	const base = path.join(appDir, "vs", "code");
	let htmlFile = path.join(base, "electron-sandbox", "workbench", "workbench.html");
        // ======= Add Support Cursor IDE
	if (!fs.existsSync(htmlFile)) {
		htmlFile = path.join(base, "electron-sandbox", "workbench", "workbench-apc-extension.html");
	}
        // =======
	if (!fs.existsSync(htmlFile)) {
		htmlFile = path.join(base, "electron-sandbox", "workbench", "workbench.esm.html");
	}
	if (!fs.existsSync(htmlFile)) {
		vscode.window.showInformationMessage(msg.unableToLocateVsCodeInstallationPath);
	}

        // ........
}
```

Then, reload window.

I hope this helps.

**Comment by mrjxtr** [comment]
Ill try to tinker around... I'm using Arch but it might work... Thanks @keyding

**Comment by be5invis** [comment]
No idea. I am not Linux expert. This _seems_ related to readonly FSes or sandboxes.

**Comment by mrjxtr** [comment]
> No idea. I am not Linux expert. This _seems_ related to readonly FSes or sandboxes.

Yeah, I am not sure how to deal with that because running `sudo chown -R $(whoami)` + the path does not seem to work.

**Comment by jameshibbard** [comment]
I'm having the same issue as @mrjxtr (Linux, running Cursor as an AppImage) and have been unable to make any of the suggestions offered in this thread work. Happy to provide details, but I am commenting here to keep the issue open.

**Comment by mrjxtr** [comment]
> I'm having the same issue as [@mrjxtr](https://github.com/mrjxtr) (Linux, running Cursor as an AppImage) and have been unable to make any of the suggestions offered in this thread work. Happy to provide details, but I am commenting here to keep the issue open.

Yeah, this should not be a thing since it is proven to be doable in vscode in linux, but I guess because since they are distributing it as an appimage, it makes it more complicated to overwrite things.

**Comment by jameshibbard** [comment]
Did you try this? 
https://github.com/getcursor/cursor/issues/913#issuecomment-1726736545

**Comment by ebetica** [comment]
Add cursor to "full disk access" under "Privacy and security" fixes it for me.

**Comment by mrjxtr** [comment]
> Add cursor to "full disk access" under "Privacy and security" fixes it for me.

This is a windows feature? The issue is in Linux

**Comment by jameshibbard** [comment]
Unfortunately "Full Disk Access" is a macOS-specific setting. It does not apply to AppImages on Linux.
